thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js

```diff
@@ -1,15 +1,229 @@
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url, postProps).then(res => resolve(checkStatus(res)));
+    });
+  },
+
+  /**
+   * Generates/updates the metrics for an existing inGraphs dashboard
+   * https://iwww.corp.linkedin.com/wiki/cf/display/ENGS/Onboarding+ingraph+dashboards
+   * @method onboardMetricsToDataset
+   * @param {String} rrd - The metric RRD
+   * @return {Ember.RSVP.Promise}
+   */
+  onboardMetricsToDataset(rrd) {
+    const postProps = {
+      method: 'post',
+      body: JSON.stringify({ RRD: rrd }),
+      headers: { 'content-type': 'Application/Json' }
+    };
+    const url = '/onboard/create';
+    return new Ember.RSVP.Promise((resolve) => {
+      fetch(url, postProps).then(res => resolve(checkStatus(res)));
+    });
+  },
+
+  /**
+   * Display results of new metrics onboarding
+   * @method prepareMetricListResult
+   * @param {Array} metricsList - List of new metrics
+   * @return {Ember.RSVP.Promise}
+   */
+  prepareMetricListResult(metricsList) {
+    if (!metricsList.Records.length) {
+      this.set('isImportError', true);
+    } else {
+      this.setProperties({
+        isImportSuccess: true,
+        importedMetrics: metricsList.Records
+      });
+    }
+  },
+
+  /**
+   * Defined actions for component
+   */
   actions: {
-    // Handles alert selection from type ahead
-    submit(data) {
-      console.log(data);
+
+    /**
+     * Reset the form... clear all important fields
+     * @method clearAll
+     * @return {undefined}
+     */
+    clearAll() {
+      this.setProperties({
+       isCustomDashFieldDisabled: false,
+       isExistingDashFieldDisabled: false,
+       isImportSuccess: false,
+       isImportError: false,
+       importExistingDashboardName: null,
+       importCustomNewDataset: null,
+       importCustomNewRrd: null,
+       datasetName: null,
+       isMetricOnboarded: false
+      });
+    },
+
+    /**
+     * Handles form submit
+     * @method submit
+     * @return {undefined}
+     */
+    submit() {
+      const isRrdImport = this.get('isExistingDashFieldDisabled');
+      const datasetName = isRrdImport ? this.get('importCustomNewDataset') : this.get('importExistingDashboardName');
+      const importObj = {
+        datasetName: datasetName,
+        dataSource: 'AutometricsThirdeyeDataSource'
+      };
+      // For either mode (import existing vs import new) make call to /onboard/create endpoint
+      this.onboardNewDataset(importObj).then(newDatasetResult => {
+        this.set('datasetName', this.get('importCustomNewDataset'));
+        // If this is a new metric from RRD, send RRD and new metric name for onboarding
+        if (isRrdImport) {
+          importObj.properties = { RRD: this.get('importCustomNewRrd') };
+          this.onboardNewDataset(importObj).then(onboardMetricsRes => {
+            // Check server for newly onboarded metrics
+            this.fetchMetricsList(datasetName).then(metricsList => {
```

Comment:
do we need to nest the .then here? 
can we `return this.fetchMetricsList` and chain the `then` on line 205?